### PR TITLE
mv: gnu test case mv-n compatibility

### PR DESCRIPTION
**Pull request by matrixhead**
fix for #6598

Behviours changed:
-  `-n' would now silently fail to copy a file when the destination already exists
- added  --update=none-fail to cp and mv
- arg parser would show a suggestion if parsing failed because of ambiguous value

### CODE REVIEW

File: src/uu/mv/src/mv.rs [hunk 1]
```diff
@@ -135,10 +135,14 @@ pub fn uumain(args: impl uucore::Args) -> UResult<()> {
...
-    if overwrite_mode == OverwriteMode::NoClobber && backup_mode != BackupMode::NoBackup {
+    if backup_mode != BackupMode::NoBackup
+        && (overwrite_mode == OverwriteMode::NoClobber
+            || update_mode == UpdateMode::ReplaceNone
+            || update_mode == UpdateMode::ReplaceNoneFail)
+    {
         return Err(UUsageError::new(
             1,
-            "options --backup and --no-clobber are mutually exclusive",
+            "cannot combine --backup with -n or --update=none-fail",
```
Comment (sylvestre) [inline src/uu/mv/src/mv.rs:145]:
```suggestion
            "cannot combine --backup with -n/--no-clobber or --update=none-fail",
```

File: tests/by-util/test_mv.rs [hunk 2]
```diff
@@ -863,14 +857,16 @@ fn test_mv_backup_off() {
...
-        .usage_error("options --backup and --no-clobber are mutually exclusive");
+fn test_mv_backup_conflicting_options() {
+    for conflicting_opt in ["--no-clobber", "--update=none-fail", "--update=none"] {
+        new_ucmd!()
+            .arg("--backup")
+            .arg(conflicting_opt)
+            .arg("file1")
+            .arg("file2")
+            .fails()
+            .usage_error("cannot combine --backup with -n or --update=none-fail");
```
Comment (sylvestre) [inline tests/by-util/test_mv.rs:868]:
```suggestion
            .usage_error("cannot combine --backup with -n/--no-clobber or --update=none-fail");
```

**Comment by matrixhead** [comment]
changes in new commits
- added `--debug` support for `mv`
- cp would fail silently when -n is given

**Comment by sylvestre** [comment]
well done and sorry for the latency

**Pull request merged**